src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,232 @@
+         IContainerTask containerTask,
+         IParameterSetUpdater parameterSetUpdater) : base(valueOriginRepository, formulaFactory, individualFactory, containerTask, parameterSetUpdater, CoreConstants.DiseaseStates.HI)
+      {
+         _parameterFactory = parameterFactory;
+         _ageDimension = dimensionRepository.AgeInYears;
+      }
+
+      public override bool ApplyTo(Individual individual) => applyTo(individual, UpdateParameter(HI_EDGINTON_VALUE_ORIGIN_ID));
+
+      public override bool ApplyForPopulationTo(Individual individual) => applyTo(individual, UpdateParameterValue);
+
+      private bool applyTo(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
+      {
+         updateBloodFlows(individual, updateParameterFunc);
+         updateVolumes(individual, updateParameterFunc);
+         updateGFR(individual, updateParameterFunc);
+         updateOntogenyFactory(individual, updateParameterFunc);
+         updateHematocrit(individual, updateParameterFunc);
+
+         addChildPughScoreToLiver(individual);
+         return true;
+      }
+
+      private void addChildPughScoreToLiver(Individual individual)
+      {
+         var organism = individual.Organism;
+         var childPughScore = organism.Parameter(CHILD_PUGH_SCORE);
+         if (childPughScore == null)
+         {
+            childPughScore = createChildPughScore();
+            organism.Add(childPughScore);
+         }
+
+         childPughScore.Value = childPughScoreFor(individual);
+      }
+
+      private IParameter createChildPughScore()
+      {
+         var childPughScore = _parameterFactory.CreateFor(CHILD_PUGH_SCORE, PKSimBuildingBlockType.Individual)
+            .WithGroup(CoreConstants.Groups.INDIVIDUAL_CHARACTERISTICS);
+         childPughScore.DefaultValue = null;
+         childPughScore.Editable = false;
+         return childPughScore;
+      }
+
+      protected override IReadOnlyList<IParameter> ParameterChangedByDiseaseStateAsList(Individual individual)
+      {
+         var organism = individual.Organism;
+         var kidney = organism.Organ(KIDNEY);
+         var liver = organism.Organ(LIVER);
+
+         var organsBloodFlow = new[] {STOMACH, SMALL_INTESTINE, LARGE_INTESTINE, SPLEEN, PANCREAS, LIVER, KIDNEY, BONE, FAT, GONADS, HEART, MUSCLE, SKIN};
+         var bloodFlows = organsBloodFlow.Select(x => organism.Organ(x).Parameter(SPECIFIC_BLOOD_FLOW_RATE)).ToList();
+
+         return new[]
+         {
+            organism.Parameter(HCT),
+            organism.Parameter(CHILD_PUGH_SCORE),
+            organism.Parameter(CHILD_PUGH_SCORE),
+            organism.Parameter(ONTOGENY_FACTOR_ALBUMIN),
+            organism.Parameter(ONTOGENY_FACTOR_AGP),
+            kidney.Parameter(GFR_SPEC),
+            liver.Parameter(VOLUME),
+         }.Union(bloodFlows).ToList();
+      }
+
+      private void updateOntogenyFactory(Individual individual, Action<IParameter, double, bool> updateParameterFunc)
```

Comment:
here and below: same comments for `updateParameterFunc `as above (bool=>enum etc.)